sdk/daml-script/test/BUILD.bazel

```diff
@@ -225,24 +225,33 @@ da_scala_library(
 )
 
 daml_compile(
-    name = "upgrade-test-lib",
+    name = "upgrade-test-lib-dev",
     srcs = [":daml/UpgradeTestLib.daml"],
     dependencies = [
         "//daml-script/daml:daml-script-2.dev.dar",
     ],
-    project_name = "upgrade-test-lib",
-    # TODO(https://github.com/digital-asset/daml/issues/18457): split the lib into modules that use
-    #  contract keys and those that don't. Revert to the default target for those that don't.
+    project_name = "upgrade-test-lib-dev",
     target = "2.dev",
     version = "1.0.0",
 )
 
+daml_compile(
+    name = "upgrade-test-lib",
+    srcs = [":daml/UpgradeTestLib.daml"],
+    dependencies = [
+        "//daml-script/daml:daml-script.dar",
+    ],
+    project_name = "upgrade-test-lib",
+    target = "2.1",
```

Comment:
Considering we use the maxStableVersion property in scala, it would probably be best to use the latest version in here too using the utils in `daml-lf/language/daml-lf.bzl`